client/alloc_runner.go

```diff
@@ -573,7 +576,19 @@ func (r *AllocRunner) syncStatus() error {
 	// Get a copy of our alloc, update status server side and sync to disk
 	alloc := r.Alloc()
 	r.updater(alloc)
-	r.allocBroadcast.Send(alloc)
+
+	// Try to send the alloc up to three times with a delay to allow recovery.
+	sent := false
+	for i := 0; i < 3; i++ {
+		if sent = r.allocBroadcast.Send(alloc); sent {
```

Comment:
@schmichael I would do the retry loop there as well. What the recovery loop is doing is allowing any listener a chance to drain the channel. The issue is the updated alloc gets dropped if the channel is full.

It shouldn't be a problem if they get duplicates. You more or less have to write idempotent code when watching on allocs.